pkg/cmd/openshift-service-serving-cert-signer/config.go

```diff
@@ -0,0 +1,214 @@
+	refs = append(refs, &config.ServingInfo.ClientCA)
+	for i := range config.ServingInfo.NamedCertificates {
+		refs = append(refs, &config.ServingInfo.NamedCertificates[i].CertFile)
+		refs = append(refs, &config.ServingInfo.NamedCertificates[i].KeyFile)
+	}
+	refs = append(refs, &config.Signer.CertFile)
+	refs = append(refs, &config.Signer.KeyFile)
+
+	return refs
+}
+
+// resolvePaths updates the given refs to be absolute paths, relative to the given base directory
+func resolvePaths(refs []*string, base string) error {
+	for _, ref := range refs {
+		// Don't resolve empty paths
+		if len(*ref) > 0 {
+			// Don't resolve absolute paths
+			if !filepath.IsAbs(*ref) {
+				*ref = filepath.Join(base, *ref)
+			}
+		}
+	}
+	return nil
+}
+
+type ClientConnectionOverrides struct {
+	// AcceptContentTypes defines the Accept header sent by clients when connecting to a server, overriding the
+	// default value of 'application/json'. This field will control all connections to the server used by a particular
+	// client.
+	AcceptContentTypes string
+	// ContentType is the content type used when sending data to the server from this client.
+	ContentType string
+
+	// QPS controls the number of queries per second allowed for this connection.
+	QPS float32
+	// Burst allows extra queries to accumulate when a client is exceeding its rate.
+	Burst int32
+}
+
+var defaultConnectionOverrides = &ClientConnectionOverrides{
+	AcceptContentTypes: "application/vnd.kubernetes.protobuf,application/json",
+	ContentType:        "application/vnd.kubernetes.protobuf",
+	QPS:                50,
+	Burst:              100,
+}
+
+// getKubeConfigOrInClusterConfig loads in-cluster config if kubeConfigFile is empty or the file if not,
+// then applies overrides.
+func getKubeConfigOrInClusterConfig(kubeConfigFile string, overrides *ClientConnectionOverrides) (*rest.Config, error) {
+	if len(kubeConfigFile) > 0 {
+		return getClientConfig(kubeConfigFile, overrides)
+	}
+
+	clientConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	applyClientConnectionOverrides(overrides, clientConfig)
+	clientConfig.WrapTransport = defaultClientTransport
+
+	return clientConfig, nil
+}
+
+func getClientConfig(kubeConfigFile string, overrides *ClientConnectionOverrides) (*rest.Config, error) {
+	kubeConfigBytes, err := ioutil.ReadFile(kubeConfigFile)
+	if err != nil {
+		return nil, err
+	}
+	kubeConfig, err := clientcmd.NewClientConfigFromBytes(kubeConfigBytes)
+	if err != nil {
+		return nil, err
+	}
+	clientConfig, err := kubeConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	applyClientConnectionOverrides(overrides, clientConfig)
+	clientConfig.WrapTransport = defaultClientTransport
+
+	return clientConfig, nil
+}
+
+// applyClientConnectionOverrides updates a kubeConfig with the overrides from the config.
+func applyClientConnectionOverrides(overrides *ClientConnectionOverrides, kubeConfig *rest.Config) {
+	if overrides == nil {
```

Comment:
Can we handle the case where `kubeConfig == nil`? It's good code hygiene 🙂IMHO.

Reply:
> Can we handle the case where kubeConfig == nil? It's good code hygiene IMHO.

We don't generally do that for wiring cases like this.  Programmer messes up, get an obvious error.